### PR TITLE
bootstrap4 ProjectsController

### PR DESCRIPTION
**Pull request by JoeCohen**
- Turn ProjectsControllerTest green, except UI issues
- Isolate assertions that fail because of incomplete UI
into their own test methods or at the end of existing methods

### CODE REVIEW

**Comment by JoeCohen** [comment]
All tests green, except for UI Fails, isolated at the end of the tests shown below.
Now for some refactoring.
```ruby
 FAIL["test_show", #<Minitest::Reporters::Suite:0x0000557763c332c0 @name="ProjectsControllerTest">, 11.906261848984286]
 test_show#ProjectsControllerTest (11.91s)
        Project page should have link to admin_request.
        Expected 0 to be >= 1.
        test/controllers/projects_controller_test.rb:87:in `test_show'

 FAIL["test_show_logged_in", #<Minitest::Reporters::Suite:0x0000557762a8f578 @name="ProjectsControllerTest">, 13.60731560899876]
 test_show_logged_in#ProjectsControllerTest (13.61s)
        Project page should have link to admin_request.
        Expected 0 to be >= 1.
        test/controllers/projects_controller_test.rb:98:in `test_show_logged_in'
```

File: app/views/projects/_form.html.erb [hunk 1]
```diff
@@ -1,11 +1,12 @@
...
-<%= form_with model: :project do |f| %>
+<% # form_tag(add_query_param(action: :new)) do %>
+<% # form_tag(add_query_param(action: :edit, id: @project.id)) do %>
+<# used by new/create, edit/update %>
+<%= form_with model: @project do |f| %>
 
   <%= f.submit button_name.t, class: "btn mx-auto mt-3" %>
 
   <%= f.hidden_field :q, value: get_query_param %>
-  <%= f.hidden_field :id, value: @project.id if action == :edit %>
```
Comment (JoeCohen) [inline app/views/projects/_form.html.erb:8]: I don't know why we need the conditional.

File: app/views/projects/show.html.erb [hunk 2]
```diff
@@ -63,6 +55,8 @@
                        controller: :observations,
                        action: :observations_for_project,
                        id: @project.id) %> |
+<%# FIXME This action now in UsersController; should be something like:
+  link_to :app_checklist.t users_checklist_path(params: { project_id: @project.id })%>
```
Comment (JoeCohen) [inline app/views/projects/show.html.erb:59]: The code suggested in this comment does not cause the test to fail.
But following the link in dev mode, throws an error.

Comment (nimmolo) [reply app/views/projects/show.html.erb:59]: Hi Joe- i don't know if it makes a difference, but it looks like syntax should be  users_checklist_path(project_id: @project.id)

File: app/controllers/projects_controller.rb [hunk 3]
```diff
@@ -130,41 +113,41 @@ def show_selected_projects(query, args = {})
...
+    @is_admin = @project.is_admin?(@user)
+
+    @draft_data = Project.connection.select_all %(
+      SELECT n.display_name, nd.id, nd.user_id
+      FROM names n, name_descriptions nd, name_descriptions_admins nda
+      WHERE nda.user_group_id = #{@project.admin_group_id}
+        AND nd.id = nda.name_description_id
+        AND n.id = nd.name_id
+      ORDER BY n.sort_name ASC, n.author ASC
+    )
```
[the remark this answers is not in the document]
Comment (nimmolo) [reply app/controllers/projects_controller.rb:132]: This seems like a question for @pellaea

**Comment by pellaea** [comment]
Ha!  I've never really learned the new ActiveRecord syntax.  This looks way
too complicated for me to translate into AR methods.  Joe is probably way
more expert at that than I!

**Pull request merged**